openchain/consensus/obcpbft/obc-sieve.go

```diff
@@ -36,29 +36,52 @@ type obcSieve struct {
 	stack consensus.Stack
 	pbft  *pbftCore
 
+	startup chan []byte
+
 	id            uint64
 	epoch         uint64
 	imminentEpoch uint64
 	blockNumber   uint64
 	currentReq    string
-	currentResult []byte
+	currentView   uint64
+	lowWaterSeqNo uint64
 
-	currentTx   []*pb.Transaction
 	verifyStore []*Verify
 
 	queuedExec map[uint64]*Execute
 	queuedTx   [][]byte
+
+	executor        Executor
+	validResultChan chan *validResult
+}
+
+type validResult struct {
+	commit  bool
+	sieveID []byte
+	peerIDs []*pb.PeerID
 }
 
 func newObcSieve(id uint64, config *viper.Viper, stack consensus.Stack) *obcSieve {
 	op := &obcSieve{stack: stack, id: id}
 	op.queuedExec = make(map[uint64]*Execute)
-	op.pbft = newPbftCore(id, config, op, stack)
-	op.pbft.sts.RegisterListener(op)
+	op.validResultChan = make(chan *validResult)
+	op.startup = make(chan []byte)
+
+	op.executor = NewOBCExecutor(config, op, stack)
+
+	logger.Debug("Replica %d obtaining startup information", id)
+	startupInfo := <-op.startup
+	close(op.startup)
+
+	op.pbft = newPbftCore(id, config, op, startupInfo)
 
 	return op
 }
 
+func (op *obcSieve) Startup(seqNo uint64, id []byte) {
```

Comment:
Why doesn't NewOBCExecutor return the values passed to Startup()?  Why do all consumers have to implement this channel init blocking thing?